File: src/main/java/com/tkyngs/hurima/model/entity/Item.java
package com.tkyngs.hurima.model.entity;

import lombok.*;

import javax.persistence.*;

/**
 * @author Sowa Takayanagi
 * @version 1.0.0
 * @since 1/8/2022 7:52 AM
 */

@Getter
@Setter
@ToString
@RequiredArgsConstructor
@Entity
@Table(name = "items")
public class Item {

  @Id
  @GeneratedValue(strategy = GenerationType.SEQUENCE)
  @SequenceGenerator(name = "items_id_seq", sequenceName = "items_id_seq", initialValue = 4, allocationSize = 1)
  private Long id;

  private String name;
  @ManyToOne
  @JoinColumn(name = "user_id")
  private User seller;
  private String sellerEmailDomain;
  private String condition;
  private String description;
  private String price;
}

File: web/src/entity/Item.ts
/**
 * @author  Sowa Takayanagi
 * @since   1/16/2022 7:58 PM
 * @version 1.0.0
 */
import {User} from "./User";


export type Item = {
  id: number
  name: string
  seller: User
  sellerEmailDomain: string
  condition: string
  description: string
  price: string
}
